storage: TestEngineKeyValidate failed

storage.TestEngineKeyValidate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11592504?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11592504?buildTab=artifacts#/) on master @ [d9c137d7883892e354b767b299bd383576d2a542](https://github.com/cockroachdb/cockroach/commits/d9c137d7883892e354b767b299bd383576d2a542):


```
=== RUN   TestEngineKeyValidate
--- FAIL: TestEngineKeyValidate (0.02s)
=== RUN   TestEngineKeyValidate/randomized
    --- FAIL: TestEngineKeyValidate/randomized (0.01s)
=== RUN   TestEngineKeyValidate/randomized/LockTableKey
    engine_key_test.go:304: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/storage/engine_key_test.go:304
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestEngineKeyValidate/randomized/LockTableKey
        --- FAIL: TestEngineKeyValidate/randomized/LockTableKey (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/storage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEngineKeyValidate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31155

I left this running under stress and eventually hit the failure:
```
--- FAIL: TestEngineKeyValidate (0.24s)
    --- FAIL: TestEngineKeyValidate/"a"/0,0 (0.00s)
        engine_key_test.go:274:
            	Error Trace:	github.com/cockroachdb/cockroach/pkg/storage/engine_key_test.go:274
            	Error:      	Not equal:
            	            	expected: 0
            	            	actual  : 1
            	Test:       	TestEngineKeyValidate/"a"/0,0
```
Notice that this was not `TestEngineKeyValidate/randomized/LockTableKey`, so it isn't specific to lock table keys.

--- a/pkg/storage/engine_key_test.go
+++ b/pkg/storage/engine_key_test.go
@@ -288,7 +288,7 @@ func TestEngineKeyValidate(t *testing.T) {
 				if err := ek.Validate(); err != nil {
 					t.Errorf("%q.Validate() = %q, want nil", ek, err)
 				}
-				require.Equal(t, 0.0, testing.AllocsPerRun(1, func() {
+				require.Equal(t, 0.0, testing.AllocsPerRun(1000, func() {
 					_ = ek.Validate()
 				}))
 			}
@@ -301,7 +301,7 @@ func TestEngineKeyValidate(t *testing.T) {
 				if err := ek.Validate(); err != nil {
 					t.Errorf("%q.Validate() = %q, want nil", ek, err)
 				}
-				require.Equal(t, 0.0, testing.AllocsPerRun(1, func() {
+				require.Equal(t, 0.0, testing.AllocsPerRun(1000, func() {
 					_ = ek.Validate()
 				}))
 				buf = ek.Key[:0]

